core-ui/src/components/ApiRules/components.js

```diff
@@ -39,7 +39,7 @@ export function GoToApiRuleDetails({ apiRule }) {
 
 export function CopiableApiRuleHost({ apiRule }) {
   const { t } = useTranslation();
-  let hostname = apiRule.spec.service.host || apiRule.spec.host;
+  let hostname = apiRule?.spec?.service?.host || apiRule?.spec?.host;
```

Comment:
According to schema, all those fields are required, so i'm not sure whether optional chaining is needed.

Reply:
It may receive empty apiRule so I removed the rest of the fixes